### ISSUE TITLE
Downloading HP Model BIOS update - URL seems to have changed and .cab naming convention.

### ISSUE THREAD

**Issue by Love-A** [maurice-daly/DriverAutomationTool#496]
Hi!

I have been having issues downloading BIOS for models using DAT.
The DAT log says:
[Error] - An error occurred while attempting contact http://ftp.hp.com/pub/caps-softpaq/cmit/imagepal/ref/8b94/8b94_64_22H2.cab - The remote server returned an error: (404) Not Found.

Checking the DAT .ps1, this is what determines the URL in version 7.2.2:
$HPXMLCabinetSource = "http://ftp.hp.com/pub/caps-softpaq/cmit/imagepal/ref/" + $($($SKUValue.Split(",") | Select-Object -First 1) + "/" + $($SKUValue.Split(",") | Select-Object -First 1) + "_" + $($Architecture.TrimStart("x")) + "_" + $OS + ".cab")

Entering the URL above, manually to browser, you get an 404.

So, i tried to find what URL could be vaild for the specific model. Using HPIA with /debug i found out that that uses this URL for the same model to download the .cab:
08-17-2023 11:55:39 --     Downloading 'https://hpia.hpcloud.hp.com/ref/8b94/8b94_64_10.0.22h2.cab?hpia=5.1.10&wf=3&pid=2101628554&osv=10.0.19045&osa=64&m=ui&at=User'

You can use just "https://hpia.hpcloud.hp.com/ref/8b94/8b94_64_10.0.22h2.cab"

With this information, it seems as if the URL in .ps1 doesnt add up anymore, and the naming convetion for the .cab has changed aswell, and would need a tweak.

**Comment by hillihappo** [comment]
@Love-A same for us. Does anyone have a way around this?

**Comment by stephannn** [comment]
is this the same issue? https://github.com/maurice-daly/DriverAutomationTool/issues/454

**Comment by hillihappo** [comment]
@stephannn I don't think so since the link doesn't seams to be working with HTTPS.

http://ftp.hp.com/pub/caps-softpaq/cmit/imagepal/ref/8b41/8b41_64_22H2.cab

**Comment by stephannn** [comment]
https://ftp.hp.com/pub/caps-softpaq/cmit/imagepal/ref/8b41/8b41_64_10.0.22h2.cab
as well as
http://ftp.hp.com/pub/caps-softpaq/cmit/imagepal/ref/8b41/8b41_64_10.0.22h2.cab
are working. The H is case sensitive
you also have to change it to the new naming convention

**Comment by hillihappo** [comment]
@stephannn ahh nice, my fault of not reading every post. I will try your solution.

**Comment by Love-A** [comment]
Thank you @stephannn ! I was looking in the Pulls but i must have missed that one!

**Comment by niwsen** [comment]
I still have the issue, but with newer BIOS versions

Any fixes for this? Tried both H and h in the url.

- URL is https://ftp.hp.com/pub/caps-softpaq/cmit/imagepal/ref/8b41/8b41_64_10.0.23h2.cab	DriverAutomationTool	2/26/2024 1:19:50 PM	23232 (0x5AC0)
======== Errors(s) Occurred ========	DriverAutomationTool	2/26/2024 1:19:51 PM	23232 (0x5AC0)
[Error] - An error occurred while attempting contact https://ftp.hp.com/pub/caps-softpaq/cmit/imagepal/ref/8b41/8b41_64_10.0.23h2.cab - The remote server returned an error: (404) Not Found.	DriverAutomationTool	2/26/2024 1:19:51 PM	23232 (0x5AC0)
 	DriverAutomationTool	2/26/2024 1:19:51 PM	23232 (0x5AC0)
- Could not download HP EliteBook 840 14 inch G10 Notebook PC BIOS update	DriverAutomationTool	2/26/2024 1:19:51 PM	23232 (0x5AC0)
Warning: Unable to retrieve BIOS Download URL For HP Client Model: EliteBook 840 14 inch G10 Notebook PC	DriverAutomationTool	2/26/2024 1:19:51 PM	23232 (0x5AC0)

**Comment by harris1m** [comment]
Is there is fix for this yet? We are on the cusp of moving from Dell to HP. I can get around the driver download by manually downloading the SP from HPIA and Copying to the DAT source folder. Ensuring you do not have "Remove driver source packages" ticked. I cannot do the same for BIOS, haven't found the site to download these yet.

**Comment by ajn142attamu** [comment]
@niwsen and @harris1m I think I've got a fix that will work for y'all in my fork. I've updated the URL to the softpaqs per @Love-A and @stephannn 's findings and added handling for 22H2 and 23H2.

You can copy https://github.com/ajn142attamu/DriverAutomationTool/blob/master/Current%20Branch/7.2.2/DriverAutomationTool.ps1 into `C:\Program Files\MSEndpointMgr\Driver Automation Tool` and then run it from an admin PowerShell window. You may need to clear the `C:\Program Files\MSEndpointMgr\Driver Automation Tool\Temp` directory and rename `C:\Program Files\MSEndpointMgr\Driver Automation Tool\Settings\OEMLinks.xml` to ensure you pull the latest copy of OEMLinks.xml and then the latest copy of all the catalog files.

**Comment by niwsen** [comment]
Thanks ajn142attamu, I will try this monday, when I'm at the customer.

Though yours is 7.2.2 ... do you have the latest from the 7.2.3 in your .ps1 file?

**Comment by harris1m** [comment]
@ajn142attamu Thank you, give me some time to test. Will get back to you all.

**Comment by harris1m** [comment]
BIOS now downloads OK. Correct HPIA URL working. 

![image](https://github.com/maurice-daly/DriverAutomationTool/assets/27958889/c205058b-cbb3-43db-820c-294ae921d780)

**Comment by ajn142attamu** [comment]
> Thanks ajn142attamu, I will try this monday, when I'm at the customer.
> 
> Though yours is 7.2.2 ... do you have the latest from the 7.2.3 in your .ps1 file?

Yeah, source code for 7.2.3 wasn’t released, and the changelog said the only change was W11-23H2 support. I’ve patched that support into my fork of the 7.2.2 source code so that I could continue being able to try and fix bugs as encountered while getting the benefits of 7.2.3.

**Comment by korpx** [comment]
https://github.com/maurice-daly/DriverAutomationTool/issues/454#issue-1560157199

**Comment by Tof006** [comment]
Hi,
I'm having a problem when trying to download HP BIOS package for W11 24H2

![Image](https://github.com/user-attachments/assets/8a5b769c-c867-47b9-9c4d-311a908346a1)
What do I need to change to be able to download the correct package?
Thanks

**Comment by mrm1432** [comment]
I've got the bios working using the ajn142attamu script, but how to I know update to the newest version, without losing the changes to the bios download configs?
